lib/reek/smells/attribute.rb

```diff
@@ -1,4 +1,6 @@
+require_relative 'smell_configuration'
```

Comment:
Right – as long as `smell_detector.rb` keeps requiring `smell_configuration.rb`. :)

Disclaimer: the blow is a bit hair-splitting in most cases, but I want to clarify my approach (sorry!).

My general rule-of-thumb is ‘if `attribute.rb` refers to `SmellConfiguration` then it’s coupled to it and should require the file that defines it’ – this goes along my earlier argument ‘if a file starts with a long list of `requires` then it’s because the implementation is too tightly coupled to too many things, and the `requires` should not be hidden to pretend it’s not’.

Or, to rephrase – if `smell_detector.rb` requires `smell_configuration.rb` and `attribute.rb` should not require `smell_configuration.rb`, then `SmellDetector` should have a method that exposes `SmellConfiguration` and `Attribute` implementation should use that method to get it (rather than calling `SmellConfiguration` explicitly).